Libraries/LibWeb/Credentials/NavigatorCredentialsContainer.cpp

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "NavigatorCredentialsContainer.h"
+
+#include <LibWeb/HTML/Navigator.h>
+
+namespace Web::Credentials {
+GC::Ref<CredentialsContainer> NavigatorCredentialsContainer::credentials()
+{
+    auto const& navigator = verify_cast<HTML::Navigator>(*this);
```

Comment:
Rather than doing this, add a method similar to this one on NavigatorStorage

```
 // ^StorageAPI::NavigatorStorage
    virtual Bindings::PlatformObject const& this_navigator_storage_object() const override { return *this; }
```
and then you can get the realm from the platform object instead of making assumptions about this.
    
For example, what happens if this API is exposed to some sort of Worker global in the future? (via WorkerNavigator)